src/customers/controller/customers.controller.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppModule } from '../../app.module';
+import { CustomersController } from './customers.controller';
+
+describe('CustomersController', () => {
+  let controller: CustomersController;
+
+  // const requestMock = {
+  //   query:{},
+  // } as unknown as Request;
+
+  // const responseMock ={
+  //   status: jest.fn(x=>{
+  //     jest.fn(y=>y)
+  //   }),
+  //   send: jest.fn(x=>x)
+  // }
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
```

Comment:
**suggestion:** Keep in mind to provide the dependencies for the service you are trying to test.
You can auto-mock the dependencies like this: https://docs.nestjs.com/fundamentals/testing#auto-mocking

Then use the spyOn property from jest to mock the specific methods in your test:
```typescript
    jest.spyOn(catsService, 'findAll').mockImplementation(() => result);

      expect(await catsController.findAll()).toBe(result);
```

Reply:
Alternativaly, you can create mock classes with your desired implementation and keep them in a shared place to be reused by your tests. Later on, you can provide them in the `providers` property in `createTestingModule` using
```
{ provide: MyClass, useValue: MyClassInstance}
```
(you also have instead of useValue other options like useClass, useFactory, etc)